aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -32,17 +48,16 @@ Tensor mkldnn_linear(
       "mkldnn_linear: input needs to has dim at least 2, input dim ", self.dim());
   TORCH_CHECK(self.is_mkldnn(),
       "mkldnn_linear: input needs to be mkldnn layout");
-  TORCH_CHECK(weight.is_mkldnn() && bias.is_mkldnn(),
```

Comment:
The backward formulas seems to actually assume that the weights are no mkldnn Tensors (because mkldnn_linear_backward_weights always convert to dense() the outputs before returning them). Is that true? If not, the backward formula should be updated to make sure it returns the same type as the weights.

Reply:
For backward path, we want to support two method.

1. input is a MKLDNN tensor, weight is  a dense tensor, this our primary method, but this method always has weight reorder at each iteration.
2.  input is a MKLDNN tensor, weight is a MKLDNN tensor by call **model.to_mkldnn()** to transfer weight to a MKLDNN tensor before training,  \this method will reduce the weight reorder time, but for the optimizer step, we may meet some ops not support MKLDNN tensor, and also have a problem when we save the model: t**orch.sav**e can't save MKLDNN tensor, we may can solve those problems to get more better performance in the future .

Reply:
Ok, So the backward formula needs to be updated to properly support case 2 right?

Reply:
Sorry for confuse you, for the first step, backward formula support case 1, I remove this check because: for inference, input , weight and bias are MKLDNN tensor, for training, input is MKLDNN tensor, weight  and bias are dense tensor.

Reply:
Sounds good.
Could you add the corresponding checks for the backward function to be sure that you have what you want. Nothing else prevent the user from doing training with weight/bias as MKLDNN Tensors right?

Reply:
Add some checks for backward function.